src/SourceGenerators/Uno.UI.SourceGenerators.netcore.Tests/MetadataUpdates/HotReloadWorkspaceProvider.cs

```diff
@@ -0,0 +1,341 @@
+			}
+			else
+			{
+				taskCompletionSource.TrySetException(new InvalidOperationException($"Failed to create MSBuildWorkspace: {diag.Diagnostic}"));
+			}
+		};
+
+		var currentSolution = workspace.CurrentSolution;
+		var frameworkReferences = BuildFrameworkReferences();
+		var references = BuildUnoReferences().Concat(frameworkReferences);
+		
+		var generatorReference = new MyGeneratorReference(
+			ImmutableArray.Create<ISourceGenerator>(new XamlGenerator.XamlCodeGenerator()));
+
+		foreach (var projectName in _additionalFiles.Keys.Concat(_sourceFiles.Keys).Distinct())
+		{
+			var projectInfo = ProjectInfo.Create(
+							ProjectId.CreateNewId(),
+							VersionStamp.Default,
+							name: projectName,
+							assemblyName: projectName,
+							language: LanguageNames.CSharp,
+							filePath: Path.Combine(_baseWorkFolder, projectName + ".csproj"),
+							outputFilePath: _baseWorkFolder,
+							metadataReferences: references,
+							compilationOptions: new CSharpCompilationOptions(
+								OutputKind.DynamicallyLinkedLibrary,
+								optimizationLevel: OptimizationLevel.Debug,
+								allowUnsafe: true,
+								nullableContextOptions: NullableContextOptions.Enable,
+								assemblyIdentityComparer: DesktopAssemblyIdentityComparer.Default),
+							analyzerReferences: new[] { generatorReference });
+
+			projectInfo = projectInfo
+				.WithCompilationOutputInfo(
+					projectInfo.CompilationOutputInfo.WithAssemblyPath(Path.Combine(_baseWorkFolder, projectName + Guid.NewGuid() + ".exe")));
+
+			var project = workspace.AddProject(projectInfo);
+			currentSolution = project.Solution;
+
+			if (_sourceFiles.TryGetValue(projectName, out var sourceFiles))
+			{
+				foreach (var (fileName, content) in sourceFiles)
+				{
+					var documentId = DocumentId.CreateNewId(project.Id);
+
+					currentSolution = currentSolution.AddDocument(
+						documentId,
+						fileName,
+						CSharpSyntaxTree.ParseText(content, encoding: Encoding.UTF8).GetText(),
+						filePath: Path.Combine(_baseWorkFolder, project.Name, fileName)
+					);
+				}
+			}
+
+			if (_additionalFiles.TryGetValue(projectName, out var additionalFiles))
+			{
+				foreach (var (fileName, content) in additionalFiles)
+				{
+					var documentId = DocumentId.CreateNewId(project.Id);
+					currentSolution = currentSolution.AddAdditionalDocument(
+						documentId,
+						fileName,
+						CSharpSyntaxTree.ParseText(content, encoding: Encoding.UTF8).GetText(),
+						filePath: Path.Combine(_baseWorkFolder, project.Name, fileName));
+				}
+
+				if (additionalFiles.Any())
+				{
+					// Build the analyzer document additional data information
+					var analyzerDocumentId = DocumentId.CreateNewId(project.Id);
+
+					var globalConfigBuilder = new StringBuilder($"""
+						is_global = true
+						# For now, there is no need to customize these for each test.
+						build_property.MSBuildProjectFullPath = C:\Project\{project.Name}.csproj
```

Comment:
Does this will work properly on unix system?

Also, I assume that the file won't be written to the disc, but if so, not sure it will work in you are not admin of your machine

Reply:
The test suite does not, currently, that's why the test job is running on windows, but we'll need to fix it to get faster builds. That is a good reason for it not to work :) (even if the path is not used)